cmplibrary/src/test/java/com/sourcepoint/cmplibrary/SourcePointClientTest.java

```diff
@@ -0,0 +1,205 @@
+
+        sourcePointClient.getGDPRStatus(onLoadComplete);
+
+        verify(onLoadComplete, times(1)).onSuccess(eq("http://google.com"));
+
+        verify(onLoadComplete, never()).onFailure(any(ConsentLibException.class));
+
+        reset(onLoadComplete);
+
+        doGetGDPRStatus(false);
+
+        sourcePointClient.getSiteID(onLoadComplete);
+
+        verify(onLoadComplete, never()).onSuccess(any());
+
+        verify(onLoadComplete, times(1)).onFailure(any(ConsentLibException.class));
+    }
+
+    private void doGetCustomConsents(final boolean onSuccess) {
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws JSONException {
+                System.out.println("TTTT : CALLED");
+                JSONObject jsonObject = new JSONObject();
+
+                SourcePointClient.ResponseHandler listener = (SourcePointClient.ResponseHandler) invocation.getArguments()[1];
+                String response = "{\"consentedPurposes\":[{\"_id\":\"5d287f273e5ba6241423f58d\",\"name\":\"Personalisation\"},{\"_id\":\"5d287f273e5ba6241423f58e\",\"name\":\"Essential Cookies\"}],\"consentedVendors\":[{\"_id\":\"5b07836aecb3fe2955eba270\",\"name\":\"Google Ad Manager\",\"vendorType\":\"CUSTOM\"}]}";
+
+                if (onSuccess) {
+                    try {
+                        jsonObject = new JSONObject(response);
+                    } catch (JSONException err) {
+                        System.out.println("Error " + err.toString());
+                    }
+                    Header[] headers = new Header[]{};
+                    listener.onSuccess(200, headers, jsonObject);
+                } else {
+                    Header[] headers = new Header[]{};
+                    listener.onFailure(404, headers, "Error", new ConsentLibException("Error"));
+                }
+
+                return null;
+            }
+        }).when(http).get(anyString(), any(ResponseHandlerInterface.class));
+    }
+
+    @Test
+    public void getCustomConsents()  {
+        doGetCustomConsents(true);
+
+        String anyString[] = {"consnetUUID", "euConsent", "siteId"};
+
+        sourcePointClient.getCustomConsents("consentUUID", "euConsent", "siteId", anyString, onLoadComplete);
+
+        ArgumentCaptor<HashSet<Consent>> captor = ArgumentCaptor.forClass(HashSet.class);
+        verify(onLoadComplete, times(1)).onSuccess(captor.capture());
+
+        HashSet<Consent> consents = captor.getValue();
+
+        for (Consent c : consents) {
+            System.out.println("TTTT " + c.id);
+        }
+
+        verify(onLoadComplete, never()).onFailure(any(ConsentLibException.class));
+
+        reset(onLoadComplete);
+
+        doGetCustomConsents(false);
+
+        sourcePointClient.getSiteID(onLoadComplete);
+
+        verify(onLoadComplete, never()).onSuccess(any());
+
+        verify(onLoadComplete, times(1)).onFailure(any(ConsentLibException.class));
+    }
+}
```

Comment:
I noticed the tests are only performed on mock objects. That's fine when we want to test other functionalities, like the `onSuccess` and `onFailure` callbacks are called and so on. 
But we need to test the real endpoint as well. That is, for example, when calling `getSiteId` for a specific account id and site name, we'll get an expected response. Maybe that's more like a "integration" test rather than unit testing. Does that make sense?